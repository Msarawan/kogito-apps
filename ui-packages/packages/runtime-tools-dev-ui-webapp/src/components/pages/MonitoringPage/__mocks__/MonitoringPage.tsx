import React from 'react';

const MockedMonitoringPage: React.FC = () => {
  return <></>;
};

export default MockedMonitoringPage;
