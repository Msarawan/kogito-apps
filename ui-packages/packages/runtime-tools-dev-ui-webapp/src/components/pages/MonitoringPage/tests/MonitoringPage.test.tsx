import React, { JSXElementConstructor } from 'react';
import { mount } from 'enzyme';
import {
  JobsDetailsModal,
  MilestoneStatus,
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
import MonitoringPage from '../MonitoringPage';
import { MemoryRouter } from 'react-router-dom';
import {
  ProcessListGatewayApi,
  ProcessListGatewayApiImpl
} from '../../../../channel/ProcessList/ProcessListGatewayApi';
import * as ProcessListContext from '../../../../channel/ProcessList/ProcessListContext';
import { act } from 'react-dom/test-utils';
import { hasUncaughtExceptionCaptureCallback } from 'process';

jest.mock('../../../containers/MonitoringContainer/MonitoringContainer');
describe('WebApp - ProcessDetailsPage tests', () => {
  const props = {
    dataIndexUrl: 'http://localhost:8000'
  };

  const data: ProcessInstance = {
    id: '8035b580-6ae4-4aa8-9ec0-e18e19809e0b',
    processId: 'travels',
    businessKey: null,
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    roles: [],
    state: ProcessInstanceState.Active,
    rootProcessInstanceId: null,
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    error: {
      nodeDefinitionId: 'a1e139d5-4e77-48c9-84ae-3459188e90433n',
      message: 'Something went wrong'
    },
    start: new Date('2019-10-22T03:40:44.089Z'),
    end: null,
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"hotel":{"address":{"city":"Berlin","country":"Germany","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
    nodes: [
      {
        nodeId: '1',
        name: 'Book Flight',
        definitionId: 'CallActivity_2',
        id: '7cdeba99-cd36-4425-980d-e59d44769a3e',
        enter: new Date('2019-10-22T04:43:01.143Z'),
        exit: new Date('2019-10-22T04:43:01.146Z'),
        type: 'SubProcessNode'
      },
      {
        nodeId: '2',
        name: 'Confirm travel',
        definitionId: 'UserTask_2',
        id: '843bd287-fb6e-4ee7-a304-ba9b430e52d8',
        enter: new Date('2019-10-22T04:43:01.148Z'),
        exit: null,
        type: 'HumanTaskNode'
      },
      {
        nodeId: '3',
        name: 'Join',
        definitionId: 'ParallelGateway_2',
        id: 'fd2e12d5-6a4b-4c75-9f31-028d3f032a95',
        enter: new Date('2019-10-22T04:43:01.148Z'),
        exit: new Date('2019-10-22T04:43:01.148Z'),
        type: 'Join'
      },
      {
        nodeId: '4',
        name: 'Book Hotel',
        definitionId: 'CallActivity_1',
        id: '7f7d74c1-78f7-49be-b5ad-8d132f46a49c',
        enter: new Date('2019-10-22T04:43:01.146Z'),
        exit: new Date('2019-10-22T04:43:01.148Z'),
        type: 'SubProcessNode'
      },
      {
        nodeId: '5',
        name: 'Book',
        definitionId: 'ParallelGateway_1',
        id: 'af0d984c-4abd-4f5c-83a8-426e6b3d102a',
        enter: new Date('2019-10-22T04:43:01.143Z'),
        exit: new Date('2019-10-22T04:43:01.146Z'),
        type: 'Split'
      },
      {
        nodeId: '6',
        name: 'Join',
        definitionId: 'ExclusiveGateway_2',
        id: 'b2761011-3043-4f48-82bd-1395bf651a91',
        enter: new Date('2019-10-22T04:43:01.143Z'),
        exit: new Date('2019-10-22T04:43:01.143Z'),
        type: 'Join'
      },
      {
        nodeId: '7',
        name: 'is visa required',
        definitionId: 'ExclusiveGateway_1',
        id: 'a91a2600-d0cd-46ff-a6c6-b3081612d1af',
        enter: new Date('2019-10-22T04:43:01.143Z'),
        exit: new Date('2019-10-22T04:43:01.143Z'),
        type: 'Split'
      },
      {
        nodeId: '8',
        name: 'Visa check',
        definitionId: 'BusinessRuleTask_1',
        id: '1baa5de4-47cc-45a8-8323-005388191e4f',
        enter: new Date('2019-10-22T04:43:01.135Z'),
        exit: new Date('2019-10-22T04:43:01.143Z'),
        type: 'RuleSetNode'
      },
      {
        nodeId: '9',
        name: 'StartProcess',
        definitionId: 'StartEvent_1',
        id: '90e5a337-1c26-4fcc-8ee2-d20e6ba2a1a3',
        enter: new Date('2019-10-22T04:43:01.135Z'),
        exit: new Date('2019-10-22T04:43:01.135Z'),
        type: 'StartNode'
      }
    ],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21e',
        processName: 'FlightBooking',
        businessKey: null
      },
      {
        id: '2d962eef-45b8-48a9-ad4e-9cde0ad6af88',
        processName: 'HotelBooking',
        businessKey: null
      }
    ]
  };

  const getProcessList = jest.fn().mockResolvedValue([data]);

  const MockProcessListGatewayApi = jest.fn<
    Pick<ProcessListGatewayApi, 'initialLoad' | 'query'>,
    []
  >(() => ({
    initialLoad: jest.fn(),
    query: getProcessList
  }));

  let gatewayApi: ProcessListGatewayApi;

  jest
    .spyOn(ProcessListContext, 'useProcessListGatewayApi')
    .mockImplementation(() => gatewayApi);

  beforeEach(() => {
    jest.clearAllMocks();
    gatewayApi = new MockProcessListGatewayApi();
  });

  it('Snapshot test with default values', async () => {
    jest.useFakeTimers();
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <MemoryRouter>
          <MonitoringPage {...props} />
        </MemoryRouter>
      );
      jest.advanceTimersByTime(1000);
    });
    wrapper = wrapper.update();
    expect(wrapper.find(MonitoringPage)).toMatchSnapshot();
  });
});
