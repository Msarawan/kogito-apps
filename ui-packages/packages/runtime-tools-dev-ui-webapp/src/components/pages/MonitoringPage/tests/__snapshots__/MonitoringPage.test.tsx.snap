// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebApp - MonitoringPage tests Snapshot test with default values 1`] = `
<MonitoringPage>
  <PageSectionHeader
    titleText="Monitoring"
  >
    <PageSection
      data-ouia-component-type="page-section-header"
      data-ouia-safe={true}
      variant="light"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
        data-ouia-component-type="page-section-header"
        data-ouia-safe={true}
      >
        <PageTitle
          title="Monitoring"
        >
          <Flex
            data-ouia-component-type="page-title"
            data-ouia-safe={true}
          >
            <div
              className="pf-l-flex"
              data-ouia-component-type="page-title"
              data-ouia-safe={true}
            >
              <FlexItem
                spacer={
                  Object {
                    "default": "spacerSm",
                  }
                }
              >
                <div
                  className="pf-m-spacer-sm"
                >
                  <Title
                    headingLevel="h1"
                    size="4xl"
                  >
                    <h1
                      className="pf-c-title pf-m-4xl"
                      data-ouia-component-id="OUIA-Generated-Title-1"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe={true}
                    >
                      Monitoring
                    </h1>
                  </Title>
                </div>
              </FlexItem>
            </div>
          </Flex>
        </PageTitle>
      </section>
    </PageSection>
  </PageSectionHeader>
  <KogitoEmptyState
    body="Loading Data"
    title="Loading"
    type={3}
  >
    <Bullseye
      data-ouia-component-type="kogito-empty-state"
      data-ouia-safe={true}
    >
      <div
        className="pf-l-bullseye"
        data-ouia-component-type="kogito-empty-state"
        data-ouia-safe={true}
      >
        <EmptyState
          variant="full"
        >
          <div
            className="pf-c-empty-state"
          >
            <div
              className="pf-c-empty-state__content"
            >
              <EmptyStateIcon
                color="var(--pf-global--info-color--100)"
                icon={[Function]}
              >
                <InfoCircleIcon
                  aria-hidden="true"
                  className="pf-c-empty-state__icon"
                  color="var(--pf-global--info-color--100)"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden="true"
                    aria-labelledby={null}
                    className="pf-c-empty-state__icon"
                    fill="var(--pf-global--info-color--100)"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    />
                  </svg>
                </InfoCircleIcon>
              </EmptyStateIcon>
              <Title
                headingLevel="h5"
                size="lg"
              >
                <h5
                  className="pf-c-title pf-m-lg"
                  data-ouia-component-id="OUIA-Generated-Title-2"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe={true}
                >
                  Loading
                </h5>
              </Title>
              <EmptyStateBody>
                <div
                  className="pf-c-empty-state__body"
                >
                  Loading Data
                </div>
              </EmptyStateBody>
            </div>
          </div>
        </EmptyState>
      </div>
    </Bullseye>
  </KogitoEmptyState>
</MonitoringPage>
`;
